torch/distributed/distributed_c10d.py

```diff
@@ -492,6 +500,7 @@ def init_process_group(backend,
             [],
             Backend.MPI,
             None,
+            pg_options=pg_options,
```

Comment:
This is creating MPI process group right, which doesn't support options? Is this codepath exercised by tests anywhere?

Reply:
Yeah make sense, probably should remove it. I was just passing in `pg_options` to `_new_process_group_helper`, and `_new_process_group_helper` only use `pg_options` for ProcessGroupGloo and NCCL, so MPI pg will just ignore pg_options.